### PR TITLE
Add docs for unumpy.

### PR DESCRIPTION
**Pull request by hameerabbasi**
@rgommers One minor comment, would you know how to make some text `monospace` in RST while also preserving the link?

### CODE REVIEW

**Comment by rgommers** [comment]
> would you know how to make some text monospace in RST while also preserving the link?

I strongly suspect that that isn't possible without resorting to inserting raw html with `.. raw::`

File: docs/unumpy_using.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Using ``unumpy``
+================
+
+``unumpy`` is an in-progress mirror of the NumPy API which allows the user
+to dynamically switch out the backend that they are using. It also allows
+auto-selection of the backend based on the arguments passed into a function.
+
+Currently, only the following functions are supported:
+
+* NumPy `universal functions <https://www.numpy.org/devdocs/reference/ufuncs.html>`_.
```
Comment (rgommers) [inline docs/unumpy_using.rst]: I think this needs a bit more text. I actually read this as "all ufuncs are supported", but after looking at the implementation I think you just mean the `numpy.ufunc` class.

File: docs/unumpy_using.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.. currentmodule:: unumpy
+
+Using ``unumpy``
+================
+
+``unumpy`` is an in-progress mirror of the NumPy API which allows the user
+to dynamically switch out the backend that they are using. It also allows
+auto-selection of the backend based on the arguments passed into a function.
+
```
Comment (rgommers) [inline docs/unumpy_using.rst]: Before listing supported functions/objects, perhaps a bit more context here. Something like: "note: currently coverage is very incomplete, we are focusing on supporting one or a few of each *kind* of object - regular functions, ufuncs, classes and methods like `np.ufunc` and `np.ufunc.reduce`, `dtypes`, etc.".

File: docs/unumpy_using.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* ufunc-based reductions
+
+  * `sum <https://docs.scipy.org/doc/numpy/reference/generated/numpy.sum.html>`_
+  * `prod <https://docs.scipy.org/doc/numpy/reference/generated/numpy.prod.html>`_
+  * `min <https://docs.scipy.org/doc/numpy/reference/generated/numpy.amin.html>`_
+  * `max <https://docs.scipy.org/doc/numpy/reference/generated/numpy.amax.html>`_
+  * `any <https://docs.scipy.org/doc/numpy/reference/generated/numpy.any.html>`_
+  * `all <https://docs.scipy.org/doc/numpy/reference/generated/numpy.all.html>`_
+
+The following backends are supported:
```
Comment (rgommers) [inline docs/unumpy_using.rst]: Can you comment here on whether all listed objects above are supported for all backends? (for example I'm not sure if PyTorch has an exact `ufunc` equivalent).

File: docs/unumpy_using.rst (hunk 1 above)
Comment (hameerabbasi) [reply docs/unumpy_using.rst]: All `ufunc`s I found in the NumPy docs ARE supported. 🙂

Comment (rgommers) [reply docs/unumpy_using.rst]: Ah, then I need to read the code better. Can you add "All" at the start of this line?

File: docs/unumpy_using.rst [hunk 4]
```diff
@@ -4,12 +4,25 @@ Using ``unumpy``
...
 
 ``unumpy`` is an in-progress mirror of the NumPy API which allows the user
-to dynamically switch out the backend that they are using. It also allows
+to dynamically switch out the backend that is used. It also allows
 auto-selection of the backend based on the arguments passed into a function.
 
+Note that currently, coverage is very incomplete. However, we have attempted
+to provide at least one of each kind of object in ``unumpy`` for
+reference. There are :obj:`ufunc` s and :obj:`ndarray` s,  which are classes,
+methods on :obj:`ufunc` such as , and also functions such as :obj:`sum` .
```
Comment (rgommers) [inline docs/unumpy_using.rst]: missing word in `such as ,`

File: docs/unumpy_using.rst [hunk 5]
```diff
@@ -22,14 +35,14 @@ Currently, only the following functions are supported:
   * `any <https://docs.scipy.org/doc/numpy/reference/generated/numpy.any.html>`_
   * `all <https://docs.scipy.org/doc/numpy/reference/generated/numpy.all.html>`_
 
-The following backends are supported:
+You can use the :obj:`uarray.set_backend` decorator to set a backend and use the
+desired backend. Note that not every backend supports every method. For example,
+PyTorch does not have an exact :obj:`ufunc` equivalent, so we dispatch. The following
```
Comment (rgommers) [inline docs/unumpy_using.rst]: "so we dispatch." --> I assume that needs to be "so we can't dispatch."

**Pull request merged**